app/eth1wrap/eth1wrap.go

```diff
@@ -0,0 +1,158 @@
+// Copyright © 2022-2024 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package eth1wrap
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	"github.com/obolnetwork/charon/app/errors"
+	erc1271 "github.com/obolnetwork/charon/app/eth1wrap/generated"
+	"github.com/obolnetwork/charon/app/expbackoff"
+)
+
+//go:generate abigen --abi=build/IERC1271.abi --pkg=erc1271 --out=generated/erc1271.go
+
+var (
+	ErrEthClientNotConnected = errors.New("eth1 client is not connected")
+	erc1271MagicValue        = [4]byte{0x16, 0x26, 0xba, 0x7e}
+)
+
+// NewEthClientRunner returns a initialized EL client runner.
+func NewEthClientRunner(addr string, ethclientFactory EthClientFactoryFn, erc1271Factory Erc1271FactoryFn) EthClientRunner {
+	return &client{
+		addr:               addr,
+		ethclientFactoryFn: ethclientFactory,
+		eth1client:         nil,
+		erc1271FactoryFn:   erc1271Factory,
+		reconnectCh:        make(chan struct{}, 1),
+	}
+}
+
+func NewDefaultEthClientRunner(addr string) EthClientRunner {
```

Comment:
1. Not test for this new function. Maybe a very simple test that does not run it, just creates a non-nil.
2. The filename(s) does not reflect the runner, maybe it should be `runner.go`?